lib/Controller/FolderController.php

```diff
@@ -139,6 +140,14 @@ public function getFolder(int $id): DataResponse {
 		return new DataResponse($this->formatFolder($folder));
 	}
 
+	private function checkFolderExists(int $id): ?DataResponse {
+		$storageId = $this->getRootFolderStorageId();
+		if ($storageId === null) {
+			return new DataResponse([], Http::STATUS_NOT_FOUND);
+		}
+		return null;
```

Comment:
Currently you only get the root folder id which should always be available. We still need to try to fetch the folder with the provided id, the following should work for that:

```suggestion
		$storageId = $this->getRootFolderStorageId();
		if ($storageId === null) {
			return new DataResponse([], Http::STATUS_NOT_FOUND);
		}
		$folder = $this->manager->getFolder($id, $storageId);
		if ($folder === false) {
			return new DataResponse([], Http::STATUS_NOT_FOUND);
		}
		return null;
```

Reply:
Other changes seem fine :)